dandi/models.py

```diff
@@ -888,19 +956,23 @@ class BareAssetMeta(CommonModel):
 class AssetMeta(BareAssetMeta, Identifiable):
     """Metadata used to describe an asset on the server."""
 
-    identifier: UUID4 = Field(readOnly=True, nskey="schema")
-
+    id: Optional[str] = Field(readOnly=True, description="Dandi asset id")
+    identifier: Optional[UUID4] = Field(readOnly=True, nskey="schema")
     # on publish or set by server
     contentUrl: Optional[List[HttpUrl]] = Field(None, readOnly=True, nskey="schema")
 
 
 class PublishedAssetMeta(AssetMeta):
+    id: str = Field(readOnly=True, description="Dandi asset id")
     publishedBy: HttpUrl = Field(
         description="The URL should contain the provenance of the publishing process.",
         readOnly=True,
         nskey="dandi",
     )  # TODO: formalize "publish" activity to at least the Actor
     datePublished: date = Field(readOnly=True, nskey="schema")
+    url: HttpUrl = Field(
+        None, readOnly=True, description="permalink to the item", nskey="schema"
+    )
```

Comment:
I think we could just add a mix-in `Published` (similar to `Identifiable`) which would provide `publishedBy`, `datePublished` and `url` and possibly even `id` -- if we can just use generic descriptions.  That would remove code duplication and make it easier to keep those attributes consistent between Dandiset and Asset.

Reply:
we could, but i don't think we are using this for anything else.

Reply:
done.